### PR TITLE
Close Issue 22: Setup boilerplate for backend

### LINKED ISSUE #22: Create and Configure Backend Boilerplate
# Issue Template

## Prerequisites

Please answer the following questions for yourself before submitting an issue.

## Type of PR

- [ ] Feature Request
- [ ] Report Bug
- [ ] Documentation Update
- [x] Other

## Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

## Expected Behavior

Please describe the behavior you are expecting

## Current Behavior

What is the current behavior?

## Failure Information (for bugs)

Please help provide information about the failure if this is a bug. If it is not a bug, please remove the rest of this template.

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. step 1
2. step 2
3. you get it...

### Failure Logs

Please include any relevant log snippets or files here.

## Check

- [x] I have followed issue creation guidelines
- [x] I have attached proper label with the issue
- [x] The issue is assigned to a assignee

### PR DESCRIPTION
**Pull request by souravdasslg**
# Pull Request Template

## Description

1. Create models and routes directory
2. Setup logger utility
3. Connected DB with Application

Fixes #22 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit Test
- [ ] Integration Test

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by souravdasslg** [comment]
@nimish-gupta @rishabh1403 @ahhda 
After my development
1. I checked out the master
2. Pulled the master
3. Check out this feature branch
4. Merge master with this branch.

Is this approach right?

File: server/app.js [hunk 1]
```diff
@@ -1,32 +1,37 @@
...
-app.set('views', path.join(__dirname, 'views'));
-app.set('view engine', 'jade');
-
-app.use(logger('dev'));
+app.use(pino);
 app.use(express.json());
 app.use(express.urlencoded({ extended: false }));
-app.use(cookieParser());
 app.use(express.static(path.join(__dirname, 'public')));
+fs.readdirSync(modelDirPath).forEach((file) => {
```
Comment (ahhda) [inline server/app.js]: Please add empty lines for splitting code into logical blocks

File: server/app.js [hunk 2]
```diff
@@ -37,5 +42,16 @@ app.use((err, req, res) => {
   res.status(err.status || 500);
   res.render('error');
 });
+mongoose.connect(process.env.DB_HOST, {
```
Comment (ahhda) [inline server/app.js]: Here as well, empty lines for splitting code into logical blocks

File: server/app.js (hunk 1 above)
Comment (souravdasslg) [reply server/app.js]: Done as suggested!

File: server/app.js (hunk 2 above)
Comment (souravdasslg) [reply server/app.js]: Done as suggested

**Pull request merged**